protocols/bftcosi/bftcosi.go

```diff
@@ -228,125 +240,94 @@ func (bft *ProtocolBFTCoSi) RegisterOnSignatureDone(fn func(*BFTSignature)) {
 	bft.onSignatureDone = fn
 }
 
-// Shutdown will close the dispatch-method if the protocol is stopped
-// before the normal termination.
+// Shutdown closes all channels in case we're done
 func (bft *ProtocolBFTCoSi) Shutdown() error {
-	bft.doneProcessing <- true
+	defer func() {
+		// In case the channels were already closed
+		recover()
```

Comment:
Again, we don't need to close channels if it's not to terminate a `select`, so can we drop this recover here and the closing a few lines below ? 
Having a `recover()` might hide other panicking stuff...